### ISSUE TITLE
Support emissary executor for Kubeflow Pipelines

### ISSUE THREAD

**Issue by ptitzler** [elyra-ai/elyra#2243]
**Is your feature request related to a problem? Please describe.**
TL;DR:

- Docker executor is the default workflow executor for Argo on KFP
- Kubernetes 1.20+ is deprecating docker as a container runtime
- Pipeline component specifications will change, requiring changes to parts of the Elyra code

See https://www.kubeflow.org/docs/components/pipelines/installation/choose-executor/ for detailed information

TODOs:

- [x] ~Impact on and roadmap for `kfp-tekton` workflow engine, which might be impacted as well~ No impact and thus nothing to do for Elyra!
- [ ] Determine whether Elyra  already _tolerates_ KFP/emissary for some scenarios
        - Tolerate: a KFP runtime environment is configured for emissary; which of the current Elyra workflows are broken? 
- [ ] Determine what supporting KFP/emissary entails
        - Based on findings of above task
- [ ] Determine what it takes to explicitly support KFP/emissary
        - Based on findings of above task

**Comment by ptitzler** [comment]
Required KFP runtime metadata changes:
- Need to [extend the schema](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/schemas/kfp.json#L70) to support a third type 'engine type'.
- The existing `Argo` option would remain the default to retain backward compatibility and imply that the cluster is configured to use Docker as workflow executor.
- Add a new `Argo with Emissary Executor` option

Required KFP processor, VPE, etc changes
- Impact of https://www.kubeflow.org/docs/components/pipelines/installation/choose-executor/#migrate-pipeline-components-to-run-on-emissary-executor
  - on generic component handling - TBD
  - on custom component handling - TBD

**Comment by ptitzler** [comment]
A pipeline component that requires migration currently triggers the following error in KFP during pipeline execution:

![image](https://user-images.githubusercontent.com/13068832/162025271-51b14b88-d679-4607-9fed-899ee8ffd864.png)

(https://argoproj.github.io/argo-workflows/workflow-executors/#emissary-emissary)

**Comment by ptitzler** [comment]
Based on further research,  I believe it would be sufficient to do the following to 'deliver' emissary support:
 - Add emissary specific requirements to https://elyra.readthedocs.io/en/latest/user_guide/best-practices-custom-pipeline-components.html
 - Extend the KFP component parser to issue a warning if a component specification does not include the `implementation.container.command` property. (The warning would indicate that this component _might_ not work if KFP is configured to use the Emissary Workflow Executor.)

User experience if we are taking this approach:
 - JL log contains warning message, which is issued when Elyra loads the component from a catalog
 - Component is loaded into cache/palette
 - User can add component to pipeline (no indication that this is potentially a 'bad' component)
 - Pipeline _execution might fail_ with the error listed above. (See below for information why it _might_ fail instead of _will fail_.

> As a general usability follow-up, we should think about how we can surface catalog/component parser issues to the user in the UI. Since these kind of issues are hidden today, users have a hard time troubleshooting unexpected behavior, such as why  components are not displayed in the VPE palette.

Other potential solutions that I've considered but discarded, along with the reasoning:
  - extend pipeline validation to catch/surface issues:
     - this requires introduction of a new engine type (argo with emissary) which the validator would have to take into consideration
     - a user would generally not be able to address the potential issue in JupyterLab/Elyra, because it might require an update to the component definition
     - this approach would not address the general issue mentioned above that there are many other issues that are not surfaced today
  - discard component definitions that are not compatible with the emissary workflow executor while the component cache is populated
     - this approach is not viable because users might have configured their Argo deployment in such a way that the component's container might be runnable, as indicated here: https://argoproj.github.io/argo-workflows/workflow-executors/#image-index
